nautobot/ipam/api/views.py

```diff
@@ -405,36 +411,86 @@ def available_vlans(self, request, pk=None):
         By default, the number of VIDs returned will be equivalent to PAGINATE_COUNT.
         An arbitrary limit (up to MAX_PAGE_SIZE, if set) may be passed, however results will not be paginated.
         """
-        obj = get_object_or_404(self.queryset, pk=pk)
+        vlan_group = get_object_or_404(self.queryset, pk=pk)
 
-        try:
-            limit = int(request.query_params.get("limit", get_settings_or_config("PAGINATE_COUNT")))
-        except ValueError:
-            limit = get_settings_or_config("PAGINATE_COUNT")
+        if request.method == "POST":
+            with cache.lock(
+                "nautobot.ipam.api.views.available_vlans", blocking_timeout=5, timeout=settings.REDIS_LOCK_TIMEOUT
+            ):
+                # Normalize to a list of objects
+                requested_vlans = request.data if isinstance(request.data, list) else [request.data]
```

Comment:
Is there a reason we're not following the available-prefixes/available-ips pattern of passing `request.data` into a `VLANAllocationSerializer` and doing an initial `serializer.is_valid(raise_exception=True)` before proceeding to the below data manipulation?

Reply:
It is not possible to validate `VLAN` instance in the same way because of missing `vid` attribute.

I think, functionally our used second serialization does the same?

Reply:
I think you're right. Wonder if we should change the approach used in the other two endpoints to follow this pattern?

Reply:
I revisited this and fixed 👌